app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -190,6 +194,75 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
     	return mt.ReadFromMongo(numberOfRecords);
     }
 
+    private static OkHttpClient httpClient = null;
+
+
+    // read from http source like cloud hosted parakeet receiver.cgi / json.get
+    public static List<TransmitterRawData> ReadHttpJson(String url, int numberOfRecords) {
+        List<TransmitterRawData> trd_list = new LinkedList<TransmitterRawData>();
+
+        try {
+
+            if (httpClient == null) {
+                httpClient = new OkHttpClient();
+                // suitable for GPRS
+                httpClient.setConnectTimeout(30, TimeUnit.SECONDS);
+                httpClient.setReadTimeout(60, TimeUnit.SECONDS);
+                httpClient.setWriteTimeout(20, TimeUnit.SECONDS);
+            }
+
+            Gson gson = new GsonBuilder().create();
+
+            // simple HTTP GET request
+            // n=numberOfRecords for backfilling
+            // r=sequence number to avoid any cache
+            // expecting json reply like the standard json server in dexterity / python pi usb / parakeet
+            Request request = new Request.Builder()
+
+                    // Mozilla header facilitates compression
+                    .header("User-Agent", "Mozilla/5.0")
+                    .header("Connection","close")
+                    .url(url + "?n=" + Integer.toString(numberOfRecords)
+                            + "&r=" + Long.toString((System.currentTimeMillis() / 1000) % 9999999))
+                    .build();
+
+            Response response = httpClient.newCall(request).execute();
+            // if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
+            if (response.isSuccessful()) {
+
+                String lines[] = response.body().string().split("\\r?\\n");
+
+                for (String data : lines) {
+
+                    if (data == null) {
+                        System.out.println("received null exiting");
+                        continue;
+                    }
+                    if (data.equals("")) {
+                        System.out.println("received \"\" exiting");
+                        continue;
+                    }
+
+                    TransmitterRawData trd = gson.fromJson(data, TransmitterRawData.class);
+                    trd.CaptureDateTime = System.currentTimeMillis() - trd.RelativeTime;
+
+                    trd_list.add(0, trd);
+                    //  System.out.println( trd.toTableString());
+                    if (trd_list.size() == numberOfRecords) {
+                        // We have the data we want, let's get out
+                        break;
+                    }
+                }
+
+                Log.i(TAG, "Success getting http json with end size: " + Integer.toString(trd_list.size()));
+            }
+
+        } catch (Exception e) {
+            Log.e(TAG, "caught HTTPException! " + e.toString());
```

Comment:
I am pretty sure it will also generate a stacktrace which is a bit wordy for a network timeout but this should be adjusted for clarity